### PR TITLE
Cherry-pick the good stuff

### PR DESCRIPTION
**Pull request by pikesley**
Conflicts:
Gemfile
Gemfile.lock
db/schema.rb

### CODE REVIEW

**Comment by pezholio** [comment]
This good to go?

**Comment by pikesley** [comment]
Will need some creds adding to Heroku before deploy

**Comment by Floppy** [comment]
Can we merge this yet?

**Comment by pikesley** [comment]
Think so

**Comment by Floppy** [comment]
Heroku doesn't have any twitter credentials in it at the moment, is that what's needed?

**Comment by pikesley** [comment]
Yes, which requires the ODI to get a phone in order to run bots

**Comment by Floppy** [comment]
Just noticed this has @pikesley's handle hardcoded in it :)

File: app/models/dataset.rb [hunk 1]
```diff
@@ -233,6 +233,17 @@ def set_owner_avatar
 
     def send_success_email
       DatasetMailer.success(self).deliver
+
+      #TwitterNotifier.success(self).tweet
+
```
Comment (Floppy) [inline app/models/dataset.rb]: Check twitter creds and user handle before doing this lot

File: app/models/dataset.rb [hunk 2]
```diff
@@ -233,6 +233,17 @@ def set_owner_avatar
...
+      twitter_client = Twitter::REST::Client.new do |config|
+        config.consumer_key        = ENV["TWITTER_CONSUMER_KEY"]
+        config.consumer_secret     = ENV["TWITTER_CONSUMER_SECRET"]
+        config.access_token        = ENV["TWITTER_TOKEN"]
+        config.access_token_secret = ENV["TWITTER_SECRET"]
+      end
+
+      twitter_client.update("Hi @pikesley, your dataset \"#{self.name}\" is now published at #{self.gh_pages_url}")
```
Comment (Floppy) [inline app/models/dataset.rb]: twitter handle is hardcoded here

**Pull request merged**